### ISSUE TITLE
Is the code not correctly implementing self attention mask in LSTransformerDecoder?

### ISSUE THREAD

**Issue by nomadlx** [bytedance/lightseq#109]
In the c++ code for `TransformerDecoderLayer`, I see only one mask operation, which is integrated into the `softmax` kernel, but in Transformer there are two types of mask operations (key padding mask and attn mask). Decoder should achieve attn mask in self-attention in order not to see the sequence on the right, but I see that the mask parameter of softmax in self-attention of decoder is set to nullptr ([code](https://github.com/bytedance/lightseq/blob/8eef387e8f1141d607ec94d4265da90dc23bf761/lightseq/training/csrc/ops/transformer_decoder_layer.cpp#L116)). Is the code not implementing these two mask operations correctly? Or is it hidden in some other code I didn't see?

In addition, I didn't find any code for position embedding in the code whether absolute position or relative position. Will the implementation of relative positions be considered later?

**Comment by nomadlx** [comment]
Sorry, I saw mask_future
